RocketChat/Rocket.Chat.ReactNative
# [FIX] Change password not working on some cases

## Proposed changes

Fix change password logic and migrate the restAPI call updateOwnBasicInfo to methodCall saveUserProfile

## Issue(s)	


## How to test or reproduce

On the profile screen try to change the password.
Possible error scenarios:
Enter the new password equal to the current one.
Confirm the password with a different password than the current one.
Enter incorrect 2fa code.

## Screenshots

## Types of changes




## Checklist



## Further comments
